docs/scripts/migrate/js-to-jsx.md

```diff
@@ -10,17 +10,56 @@ This should usually be run from the root directory of a project.
+
+```js
+jest.mock('../../ItemHeader/DeleteItemButton.js')
+```
+
+These tend to be few in number and easy to identify, so shouldn't pose much of a problem to change.
+
+## Details
+
+### Crawling files
+
 By default, this will crawl through each `.js` file in the `src` directory (using the glob `src/**/*.js`), look for JSX syntax in the file, then rename the file to use a `.jsx` extension if appropriate.
 
-Then, it will crawl through all `.js` _and_ `.jsx` files in `src` and update file imports to match the newly renamed files. **By default, this will update imports without a file extension**, e.g. `import Component from './Component'` => `import Component from './Component.jsx'`. This is because, in testing, updating files to `.jsx` extensions without updating the imports ends up causing linting errors. Functionally, the app will still work without extensions on imports though; Vite handles it. **If you don't want to update imports without extensions**, you can use the `--skipUpdatingImportsWithoutExtension` flag when running this script. Imports that use a `.js` extension will be updated to `.jsx` either way.
+The script uses the `src/**/*.js` glob by default. If you want to crawl different directories, for example to migrate smaller pieces of a project at a time, you can specify a custom glob when running the script using the [`--globString` option](#--globstring).
 
-Lastly, the script will check `d2.config.js` in the CWD for entry points to update if the respective files have been renamed.
+### Updating imports
 
-## Tips
+The script will then crawl through all `.js` _and_ `.jsx` files in `src` (or according to the provided glob string) and update file imports to match the newly renamed files.
 
-This may update a _lot_ of files; be prepared with your source control to undo changes if needed. In VSCode, for example, there is a feature in the Source Control UI to "Discard All Changes" from unstaged files. Before running the script, stage the files you want to keep around, then run the script. If the outcome isn't what you want, you can use the "Discard All Changes" option to undo them easily.
+**By default, this will update imports without a file extension**, e.g. `import Component from './Component'` => `import Component from './Component.jsx'`. This is because, in testing, updating files to `.jsx` extensions without updating the imports ends up causing linting errors. Functionally, the app will still work without extensions on imports though; Vite handles it. **If you don't want to update imports without extensions**, you can use the [`--skipUpdatingImportsWithoutExtension` flag](#--skipupdatingimportswithoutextension) when running this script.
 
-Note that renamed files are only kept track of during script execution. If, for example, you run the script, then you want to redo it with the `--skipUpdatingImportsWithoutExtension` flag, it's best to undo all the renamed files before running the script again.
+### Updating snapshot files
```

Comment:
maybe we can also mention that there might be small changes to the snapshots themselves as a result of the React 18 upgrade (spaces, and css modules names as far as I can tell) that needs to be fixed manually (i.e. update the snapshots and verify the changes are minimal and make sense)

Reply:
Added in both pages 🙂